ddp-workspace/projects/ddp-sdk/src/lib/components/address/addressEmbedded.component.ts

```diff
@@ -277,38 +287,40 @@ export class AddressEmbeddedComponent implements OnDestroy, OnInit {
     );
 
     // derived observables
-    this.isReadOnly$ = this.state$.pipe(
-        map(state => state.isReadOnly || state.isTemporarilyDisabled),
-        distinctUntilChanged(),
-        shareReplay()
-    );
-
-    this.formErrorMessages$ = this.state$.pipe(
-        pluck('formErrorMessages'),
-        distinctUntilChanged((x, y) => util.isEqual(x, y)),
-        shareReplay()
-    );
+    // this.isReadOnly$ = this.state$.pipe(
+    //     map(state => state.isReadOnly || state.isTemporarilyDisabled),
+    //     distinctUntilChanged(),
+    //     shareReplay()
+    // );
+
+      this.isReadOnly$ = this.state$.pipe(
+          extract(state => state.isReadOnly || state.isTemporarilyDisabled)
+      );
+
+    // this.formErrorMessages$ = this.state$.pipe(
+    //     pluck('formErrorMessages'),
+    //     distinctUntilChanged((x, y) => util.isEqual(x, y)),
+    //     shareReplay()
+    // );
+      this.formErrorMessages$ = this.state$.pipe(
+          extract('formErrorMessages')
+      );
 
```

Comment:
Refactored some of this code that looked repetive to use a new custom operator rxjs `extract`. Code for that is further down. Should not introduce changes in functionality or performance. Just less code.